system/src/commonMain/kotlin/com/splendo/kaluga/system/network/services/NetworkManagerService.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network.services
+
+import com.splendo.kaluga.system.network.Network
+
+interface NetworkManagerHandler {
```

Comment:
Used only in iOS? can probably live there.

Reply:
3d4f5cd eeb4af3

Reply:
I looked a bit more at what this does, it's not terrible like this but I would recommend making a sealed class with base class with methods that are overridden by the subclasses instead of this interface. 

Implementation specific variables then only live in the subclasses, while you can pass around the super class.

Mostly this will make it more readable and avoid having a bunch or lateinits that never init if I remember that part of the code well (this actually is a little bad)

Reply:
239a1c9 6dc058b 8f08788 0669038